untrusted/build.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+
+//! Builds the FFI bindings for the untrusted side of the Intel SGX SDK
+extern crate bindgen;
+use cargo_emit::{rustc_link_arg, rustc_link_search, warning};
+use std::{env, path::PathBuf};
+
+static DEFAULT_SGX_SDK_PATH: &str = "/opt/intel/sgxsdk";
+
+fn sgx_library_path() -> String {
+    env::var("SGX_SDK").unwrap_or_else(|_| DEFAULT_SGX_SDK_PATH.into())
+}
+
+fn sgx_library_suffix() -> String {
+    let mode = env::var("SGX_MODE").unwrap_or_else(|_| String::from("SW"));
+    let suffix = match mode.as_str() {
+        "SW" => "_sim",
+        "HW" => "",
+        mode => {
+            warning!(
+                "'SGX_MODE' was set to '{}'. Should be one of 'SW' or 'HW', defaulting to 'SW'",
+                mode
+            );
+            "_sim"
+        }
+    };
+    String::from(suffix)
+}
```

Comment:
This should be decided by a "sim" feature.

Reply:
In that case should the feature be `sim` and the default be `[]`?
For getting one started in most environments `sim` seems _nice_, but the main intent is to run on hardware SGX.
Defualting to hardware in a cargo configuration is less likely to trip up multiple users and would only affect initial integrators.  Still might be a bit frustrating for integrators...

Reply:
Updated, added a `sw` feature that is part of the default.  The name was chosen to match the value stored in `SGX_MODE`